hazelcast/src/main/java/com/hazelcast/projection/impl/IdentityProjection.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.projection.impl;
+
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
+import com.hazelcast.projection.Projection;
+
+public final class IdentityProjection<I> extends Projection<I, I> implements IdentifiedDataSerializable {
```

Comment:
Can you add some javadoc? Also, you need to document (on the class level as well as the factory methods) that this class should not be used  when rolling up from 3.9 to 3.10 as the 3.9 members do not contain the class.

Reply:
This is class in `impl` package. It's documented in `Projections.identity()`. And there is also `@since 3.10`.

Reply:
It's accessible from the `Projections` class which is public API and putting the `@since` tag doesn't necessarily mean users shouldn't use it when rolling up from 3.9 to 3.10